lesson-7/server/e2e-test/e2e.js

```diff
@@ -0,0 +1,25 @@
+const request = require('supertest');
+const chai = require('chai');
+
+const todosSchema = require('./todosSchema.json');
+
+chai.use(require('chai-json-schema'));
+const url = 'http://localhost:8080';
+
+describe('invoke /todos resource', () => {
+    it('should return a payload and a successful status code', done => {
+      request(url)
+        .get(`/todos`)
+        .expect(200)
+        .end((err, res) => {
+          if (err) {
+            return done(err);
```

Comment:
This can be simpler if:

- You return the Promise: mocha can handle Promises out of the box. At the same time you get rid of `done`
- Use .then instead of .end, because .end has a [special function in supertest](https://github.com/visionmedia/supertest#example):
>If you are using the .end() method .expect() assertions that fail will not throw - they will return the assertion as an error to the .end() callback.

So your test will look like this:

```javascript
it('should return a payload and a successful status code', () => 
  request(url)
    .get(`/todos`)
    .expect(200)
    .expect('content-type', 'application/json; charset=utf-8')
    .then((res) => {
      chai.expect(res.body).to.not.be.undefined;
      chai.expect(res.body).to.be.jsonSchema(todosSchema);
    });
);
```